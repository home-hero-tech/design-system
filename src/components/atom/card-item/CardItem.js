import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import classNames from 'classnames';

import css from './CardItem.module.scss';

const ListItem = ({ primary, label, format, value }) => {
  const classes = classNames({
    [css['c-card-item']]: true,
    [css['c-card-item--primary']]: primary
  });
  return (
    <li className={classes}>
      <span>{label}</span>
      <span>
        {value instanceof Date ? moment(value).format(format) : value}
      </span>
    </li>
  );
};

ListItem.propTypes = {
  format: PropTypes.string,
  label: PropTypes.string,
  primary: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)])
    .isRequired
};

ListItem.defaultProps = {
  label: null,
  primary: false,
  format: 'DD MMM YYYY [Ã s] HH:mm'
};

export default ListItem;
