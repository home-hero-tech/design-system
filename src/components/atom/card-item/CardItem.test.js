import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CardItem from './CardItem';

describe('CardItem', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(<CardItem label="Label" value="Value" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
