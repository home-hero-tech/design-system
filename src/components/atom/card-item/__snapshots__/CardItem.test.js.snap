// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem should match with the snapshot 1`] = `
<li
  className="c-card-item"
>
  <span>
    Label
  </span>
  <span>
    Value
  </span>
</li>
`;
